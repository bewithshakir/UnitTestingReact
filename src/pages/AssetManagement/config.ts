import { IDynamicFilterProps } from "../../components/UIComponents/RightInfoPanel/DynamicFilterContent.component";

export const sortByOptions = [
    "assetManagement.sortBy.payment in progress",
    "assetManagement.sortBy.recently added lots"
  ];


export const filterByFields: IDynamicFilterProps['fields'] = [
    { name: 'state', label: 'customer-filter-panel.state', fieldType: 'select', optionUrlKey: 'assetFilter', optionAPIResponseKey: 'states', initialValue: [] },
    { name: 'city', label: 'customer-filter-panel.city', fieldType: 'select', optionUrlKey: 'assetFilter', optionAPIResponseKey: 'cities', initialValue: [] }
  ];