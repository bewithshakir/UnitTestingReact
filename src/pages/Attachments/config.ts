export const config = {
    maxAllowedFileSizeBtyes : 25000000,
    acceptedFiles: '.pdf,.docx,.xlsx',
    noOfFilesLimit: 1,
    multipleAllowed: false
};