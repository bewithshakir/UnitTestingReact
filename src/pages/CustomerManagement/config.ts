import { IDynamicFilterProps } from "../../components/UIComponents/RightInfoPanel/DynamicFilterContent.component";

export const sortByOptions = [
  "customerManagement.sortBy.atoz",
  "customerManagement.sortBy.ztoa",
  "customerManagement.sortBy.newestToOldest",
  "customerManagement.sortBy.oldestToNew",
];

export const filterByFields: IDynamicFilterProps['fields'] = [
  { name: 'date', label: 'Date', fieldType: 'dateRange', initialValue: [null, null] },
  { name: 'state', label: 'customer-filter-panel.state', fieldType: 'select', optionUrlKey: 'customerFilter', optionAPIResponseKey: 'states', initialValue: [] },
  { name: 'city', label: 'customer-filter-panel.city', fieldType: 'select', optionUrlKey: 'customerFilter', optionAPIResponseKey: 'cities', initialValue: [] },
  { name: 'paymentType', label: 'customer-filter-panel.settlement type', fieldType: 'select', optionUrlKey: 'customerFilter', optionAPIResponseKey: 'settlementType', initialValue: [] },
];