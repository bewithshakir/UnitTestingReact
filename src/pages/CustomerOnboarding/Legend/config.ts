export const config = [
    {
        index: 1,
        label: "legend.customerProfile",
        secondaryText: false,
        to: "/customer/addCustomer"
    },
    {
        index: 2,
        label: "legend.parkingLots",
        secondaryText: true,
        to: "/customer/parkingLots"
    },
    {
        index: 3,
        label: "legend.vehicles",
        secondaryText: true,
        to: "/customer/vehicles"
    },
    {
        index: 4,
        label: "legend.dsp",
        secondaryText: true,
        to: "/customer/dsps"
    },
    {
        index: 5,
        label: "legend.users",
        secondaryText: true,
        to: "/users"
    },
    {
        index: 6,
        label: "legend.attachments",
        secondaryText: true,
        to: '/customer/attachments'
    }
];

export const boxSystem = {
    border: 1
};

