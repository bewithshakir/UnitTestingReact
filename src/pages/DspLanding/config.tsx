

export const filterByFields = [];



export const SORTBY_TYPES = {
    DSP_NAME_AZ: "DSP Name A-Z",
    DSP_NAME_ZA: "DSP Name Z-A",
  };

export const DSPLanding = {
    LandingPage: {
      SortByOptions: [
        "dsp.sortBy.dspname_atoz",
        "dsp.sortBy.dspname_ztoa",
      ]
    }
  };
