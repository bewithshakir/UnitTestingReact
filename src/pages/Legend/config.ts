export const config = [
    {
        index:1,
        label: "legend.customerProfile"
    },
    {
        index:2,
        label: "legend.parkingLots"
    },
    {
        index:3,
        label: "legend.vehicles"
    },
    {
        index:4,
        label: "legend.users"
    },
    {
        index:5,
        label: "legend.dsp"
    },
    {
        index:6,
        label: "legend.attachments"
    }
]

export const boxSystem = {
    width: 276, 
    height: 375, 
    border: 1
}
