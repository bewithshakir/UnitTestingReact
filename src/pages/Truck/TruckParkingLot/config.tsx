export const sortByOptions = [
    "parkingLot.sortBy.atoz",
    "parkingLot.sortBy.ztoa"
];