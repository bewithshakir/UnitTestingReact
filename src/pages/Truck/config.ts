export const ROW_ACTION_TYPES = {
  EDIT: 'edit'
};

export const SORTBY_TYPES = {
  TRUCK_NAME_AZ: "Truck Name A-Z",
  TRUCK_NAME_ZA: "Truck Name Z-A",
};


export const TruckManagement = {
  LandingPage: {
    SortByOptions: [
      "addTruckFormLabels.sortBy.truckname_atoz",
      "addTruckFormLabels.sortBy.truckname_ztoa",
    ],
    RowActionsOptions: [
      {
        label: "menus.data-grid-actions.edit",
        action: ROW_ACTION_TYPES.EDIT
      }
    ],
    DataGridFields: {
      "TRUCKNAME": {
        field: "deliveryVehicleNm", label: "TRUCK NAME"
      },
      "LICENSE": {
        field: "licenceNo", label: "LICENSE"
      },
      "VIN": {
        field: "vinNo", label: "VIN"
      },
      "MAKEORMODEL": {
        field: "makeAndModelNm", label: "MAKE/MODEL"
      },
      "COLOR": {
        field: "colorNm", label: "COLOR"
      },
      "YEAR": {
        field: "registrationYr", label: "YEAR"
      },
      "STATUS": {
        field: "activeInactiveInd", label: "STATUS"
      },
      "DRIVERS": {
        field: "drivers", label: "DRIVERS"
      },
      "LOCATIONS": {
        field: "parkingLocationCount", label: "LOCATIONS"
      },
      "TANKS": {
        field: "tankCount", label: "TANKS"
      },
      "ORDERS": {
        field: "orderCount", label: "ORDERS"
      },
    },
  }
};
