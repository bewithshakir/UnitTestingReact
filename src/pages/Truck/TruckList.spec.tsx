import { waitFor } from '@testing-library/react';
import { renderWithClient } from '../../tests/utils';
import TruckLandingContent from './index';
import { TruckManagement } from './config';
import SortbyMenu from '../../components/UIComponents/Menu/SortbyMenu.component';
import { shallow } from 'enzyme';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual("react-router-dom") as any,
    useNavigate: () => ({
        navigate: ()=> ({
        to: '/trucks'
        })
    }),
}));


describe('renders TruckLanding page component', ()=> {
    it('renders  TruckLanding page component with all ui components', ()=> {
        const result = renderWithClient(<TruckLandingContent version="Breadcrumbs-Single" />);
        const filterBtn = result.getByTestId('filter');
        const sortBtv = result.getByRole('button', {name: /sortby menu list/i});
        const searchInput = result.getByRole('textbox', { name: /search/i });
        const addBtn = result.getByTestId('addBtn');

        expect(filterBtn).toBeInTheDocument();
        expect(sortBtv).toBeInTheDocument();
        expect(searchInput).toBeInTheDocument();
        expect(addBtn).toBeInTheDocument();
    });
    it('render data on success response', async ()=> {
        const result = renderWithClient(<TruckLandingContent version="Breadcrumbs-Single" />);
        await waitFor(()=> {
            expect(result.getByText(/John/i)).toBeInTheDocument();
        });
        
    });
});

describe('Given Sortby Menu on FuelTax Landing Page-', () => {
    const { SortByOptions } = TruckManagement.LandingPage;
    test('Render Sortby Menu', () => {
        const FuelTaxSortbyMenu = shallow(
            <SortbyMenu
                options={SortByOptions.map((sortByItem) => (sortByItem))}
                onSelect={() => jest.fn()}
            />
        );
        expect(FuelTaxSortbyMenu).toMatchSnapshot();
    });

    test('Sortby Menu Menu With Options', () => {
        const FuelTaxSortbyMenu = shallow(
            <SortbyMenu
                options={SortByOptions.map((sortByItem) => (sortByItem))}
                onSelect={() => jest.fn()}
            />
        );
        FuelTaxSortbyMenu.find(".btn-sortby").simulate('click');
        expect(FuelTaxSortbyMenu.find(".btn-sortby").hasClass('active')).toBe(true);
        expect(FuelTaxSortbyMenu.find('.sortby-popper').exists()).toBe(true);
    });
});