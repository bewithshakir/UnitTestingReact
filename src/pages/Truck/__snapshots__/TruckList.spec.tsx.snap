// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Search Input on Truck Overview Page Render Search Input onChange 1`] = `
<ForwardRef(OutlinedInput)
  className="searchinput"
  id="SearchInput"
  inputProps={
    Object {
      "aria-label": "search",
    }
  }
  onChange={[Function]}
  placeholder="Search"
  startAdornment={
    <ForwardRef(InputAdornment)
      className="adornment"
      position="start"
    >
      <ForwardRef(IconButton)
        aria-label="search"
        edge="start"
      >
        <DropSearchIcon />
      </ForwardRef(IconButton)>
    </ForwardRef(InputAdornment)>
  }
  style={
    Object {
      "width": "100%",
    }
  }
  type="text"
  value=""
/>
`;

exports[`Given Sortby Menu on Truck List Landing Page- Render Sortby Menu 1`] = `
<div>
  <WithStyles(ForwardRef(Button))
    aria-haspopup="true"
    aria-label="sortby menu list"
    className="btn-sortby "
    onClick={[Function]}
    startIcon={<Memo />}
  >
    buttons.sort by
  </WithStyles(ForwardRef(Button))>
  <ForwardRef(Popper)
    anchorEl={null}
    className="sortby-popper"
    data-testid="sortby-dropdown"
    disablePortal={true}
    open={false}
    placement="bottom-start"
    transition={true}
  >
    <Component />
  </ForwardRef(Popper)>
</div>
`;
